app/src/main/kotlin/com/simplemobiletools/smsmessenger/helpers/MessagesReader.kt

```diff
@@ -9,49 +9,59 @@ import android.util.Base64
 import com.simplemobiletools.commons.extensions.*
 import com.simplemobiletools.commons.helpers.isQPlus
 import com.simplemobiletools.commons.helpers.isRPlus
-import com.simplemobiletools.smsmessenger.models.MmsAddress
-import com.simplemobiletools.smsmessenger.models.MmsBackup
-import com.simplemobiletools.smsmessenger.models.MmsPart
-import com.simplemobiletools.smsmessenger.models.SmsBackup
+import com.simplemobiletools.smsmessenger.extensions.getConversationIds
+import com.simplemobiletools.smsmessenger.models.*
 import java.io.IOException
 import java.io.InputStream
 
 class MessagesReader(private val context: Context) {
-    fun forEachSms(threadId: Long, block: (SmsBackup) -> Unit) {
+
+    fun getMessagesToExport(
+        getSms: Boolean,
+        getMms: Boolean,
+        callback: (messages: List<MessagesBackup>) -> Unit
+    ) {
+        val conversationIds = context.getConversationIds()
+        var smsMessages = listOf<SmsBackup>()
+        var mmsMessages = listOf<MmsBackup>()
+
+        if (getSms) {
+            smsMessages = getSmsMessages(conversationIds)
+        }
+        if (getMms) {
+            mmsMessages = getMmsMessages(conversationIds)
+        }
+        callback(smsMessages + mmsMessages)
+    }
+
+    private fun getSmsMessages(threadIds: List<Long>): List<SmsBackup> {
         val projection = arrayOf(
-            Sms.SUBSCRIPTION_ID,
-            Sms.ADDRESS,
-            Sms.BODY,
-            Sms.DATE,
-            Sms.DATE_SENT,
-            Sms.LOCKED,
-            Sms.PROTOCOL,
-            Sms.READ,
-            Sms.STATUS,
-            Sms.TYPE,
-            Sms.SERVICE_CENTER
+            Sms.SUBSCRIPTION_ID, Sms.ADDRESS, Sms.BODY, Sms.DATE, Sms.DATE_SENT, Sms.LOCKED, Sms.PROTOCOL, Sms.READ, Sms.STATUS, Sms.TYPE, Sms.SERVICE_CENTER
```

Comment:
keep this the old way too split by lines..

Reply:
Yep, code reformat is doing it🫤